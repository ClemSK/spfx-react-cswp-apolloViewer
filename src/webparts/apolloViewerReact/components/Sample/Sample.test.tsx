import * as React from 'react';
import 'jest';
import 'ts-jest';
// import { createRenderer } from 'react-test-renderer/shallow';
import { Sample, ISampleProps } from './Sample';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

test('should render Sample component correctly', () => {
  // const renderer = createRenderer();
  // renderer.render(<Sample />);
  // console.log(renderer.getRenderOutput());
  // expect(renderer.getRenderOutput()).toMatchSnapshot();

  const wrapper = shallow(<Sample />);
  expect(wrapper.find('span').text()).toBe('Hello Mundo:');
  expect(wrapper.find('li').length).toBe(3);
  expect(toJson(wrapper)).toMatchSnapshot();
});
