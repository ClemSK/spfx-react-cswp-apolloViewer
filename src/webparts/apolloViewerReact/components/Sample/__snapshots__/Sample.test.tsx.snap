// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sample component correctly 1`] = `
<div>
  <span>
    Hello Mundo:
  </span>
  <ul>
    <li>
      one
    </li>
    <li>
      two
    </li>
    <li>
      three
    </li>
  </ul>
</div>
`;
